service/matching/matching_engine.go

```diff
@@ -1972,6 +1975,42 @@ func (e *matchingEngineImpl) SyncDeploymentUserData(
 					changed = true
 					deploymentData.Versions = append(deploymentData.Versions[:idx], deploymentData.Versions[idx+1:]...)
 				}
+			} else {
+
+				// Only initialize DeploymentsData if we're using the new format
+				if deploymentData.GetDeploymentsData() == nil {
+					deploymentData.DeploymentsData = make(map[string]*persistencespb.WorkerDeploymentData)
+				}
+				if deploymentData.GetDeploymentsData()[req.GetDeploymentName()] == nil {
+					deploymentData.GetDeploymentsData()[req.GetDeploymentName()] = &persistencespb.WorkerDeploymentData{}
+				}
+
+				rc := req.GetUpdateRoutingConfig()
+				tqWorkerDeploymentData := deploymentData.GetDeploymentsData()[req.GetDeploymentName()]
+
+				applyUpdatesToDeploymentData := rc == nil
+				if rc != nil {
+					if rc.GetRevisionNumber() >= tqWorkerDeploymentData.GetRoutingConfig().GetRevisionNumber() {
+						changed = true
+						// Update routing config when newer or equal revision is provided
+						tqWorkerDeploymentData.RoutingConfig = rc
+						applyUpdatesToDeploymentData = true
+					}
+				}
+
+				if applyUpdatesToDeploymentData {
```

Comment:
It seems I forgot to remove these comments from the proto files: 
> Ignored if `update_routing_config` is present and has an outdated revision number.

Based on our last discussion the req.GetUpsertVersionsData() and req.GetForgetVersions() should be applied regardless of the routing config situation